### PR TITLE
feat(Store): remove redux store dep, refactoring, add hooks and cleanup

### PR DESCRIPTION
**Pull request by amen-souissi**
- [X] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [X] Make sure you are making a pull request against the **master branch** (left side). Also you should start _your branch_ off _our master_.
- [X] Check the commit's or even all commits' message styles matches our requested structure.
- [X] Check your code additions will pass linting.

### Description
remove redux store dep, refactoring, add hooks and cleanup

### CODE REVIEW

File: src/hooks.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  const { store } = React.useContext(MoonContext);
+  //@ts-ignore can't be undefined
+  const queryResult = store.readQuery<Data>(queryId).data;
+  const [state, setState] = React.useState<Data | undefined>(queryResult);
+  //@ts-ignore can't be undefined
+  store?.subscribeToQueryResult(queryId, setState);
+
+  React.useEffect(() => {
+    //@ts-ignore can't be undefined
+    return () => store?.unsubscribeFromQueryResult(queryId, setState);
```
Comment (hyacintheknobloch) [inline src/hooks.ts]: A quoi ça sert le `?`

Comment (amen-souissi) [reply src/hooks.ts]: store &&  store.unsubscribeFromQueryResult(queryId, setState)

File: src/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  //@ts-ignore can't be undefined
+  const queryState = store.readQuery(queryId);
+  const [state, setState] = React.useState<QueryState<Data>>(queryState);
+  //@ts-ignore can't be undefined
+  store.subscribeToQuery(queryId, setState);
+
+  React.useEffect(() => {
+    //@ts-ignore can't be undefined
+    return () => store.unsubscribeFromQuery(queryId, setState);
+  }, [queryId]);
```
Comment (hyacintheknobloch) [inline src/hooks.ts]: On peut pas simplifier ?
```
//@ts-ignore can't be undefined
React.useEffect(() =>  store.unsubscribeFromQuery(queryId, setState), [queryId]);
```

File: src/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
+import * as React from "react";
+
+import { MoonContext } from "./moon-provider";
+import { QueryState, QieriesStates, QueriesResults } from "./store";
```
Comment (hyacintheknobloch) [inline src/hooks.ts]: QieriesStates => QueriesStates

File: src/moon-client.tsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  constructor(links: ILink[]) {
+    this.clients = getClients(links);
+  }
+
+  public query = async (
+    source: string,
+    endPoint: string,
+    variables: any = {},
+    deserialize?: (response: any) => any,
+    options: AxiosRequestConfig = {}
```
Comment (hyacintheknobloch) [inline src/moon-client.tsx]: move options argument before deserialize ?

File: src/query-hook.tsx [hunk 5]
```diff
@@ -115,38 +75,55 @@ export default function useQuery<QueryData = any, QueryVariables = any, Deserial
...
     // @ts-ignore can't be undefined
     return () => cancelSourceRef.current.cancel();
-  }, [value, endPoint, source]);
+  }, [currentVariables, endPoint, source]);
 
   const cancel = () => {
     if (cancelSourceRef.current) {
       cancelSourceRef.current.cancel();
-      setState({ ...state, loading: false, networkStatus: MoonNetworkStatus.Finished });
+      //@ts-ignore
```
Comment (hyacintheknobloch) [inline src/query-hook.tsx]: explain why ts ignore

File: src/hooks.ts (hunk 2 above)
Comment (amen-souissi) [reply src/hooks.ts]: nop ()=>store.unsubscribeFromQuery(queryId, setState) is executed "on unmount"

File: src/store.ts [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+  state: QueryState<Data>;
+  onStateChangeHandlers: OnStateChangeHandler<Data>[];
+  onResultChangeHandlers: OnResultChangeHandler<Data>[];
+}
+
+export interface Queries<Datas = any> {
+  [queryId: string]: Query<Datas>;
+}
+
+export interface QieriesStates<Datas = any> {
```
Comment (hyacintheknobloch) [inline src/store.ts]: QueriesStates

File: test/integration/moon-provider.test.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import * as React from "react";
+import { render, wait } from "@testing-library/react";
+
+import MoonProvider, { withMoon, IMoonContextValue } from "../../src/moon-provider";
+
+import { links } from "../moon-client.test";
+import { mockAxiosClientConstructor, AxiosClient } from "../testUtils";
+
+const MyComponent: React.FunctionComponent<IMoonContextValue> = ({ client }) => {
+  const [response, setRespons] = React.useState<any>(null);
```
Comment (hyacintheknobloch) [inline test/integration/moon-provider.test.tsx]: setResponse

**Comment by amen-souissi** [comment]
merci @hyacintheknobloch

**Pull request merged**